src/opm/io/eclipse/EclFile.cpp

```diff
@@ -53,10 +53,18 @@ void EclFile::load(bool preload) {
         std::int64_t num;
         int sizeOfElement;
 
-        if (formatted) {
-            readFormattedHeader(fileH,arrName,num,arrType, sizeOfElement);
-        } else {
-            readBinaryHeader(fileH,arrName,num, arrType, sizeOfElement);
+        try {
+            if (formatted) {
+                readFormattedHeader(fileH,arrName,num,arrType, sizeOfElement);
+            } else {
+                readBinaryHeader(fileH,arrName,num, arrType, sizeOfElement);
+            }
+        } catch (const std::exception& e){
+            OPM_THROW(std::runtime_error,
+                fmt::format("Can not open EclFile: {}. Please check if the file is corrupt", this->inputFilename));
```

Comment:
The EclFile class name is not something a user would relate to, suggest changing to "Can not open eclipse output format file: ...". Also, it may be useful to give some guidance here, for example "This may be caused by trying to run the program in parallel using MPI, using an executable that was not configured with MPI when created.", or maybe that could be done on a higher level.

Reply:
Good time to also change typos. Shouldn't this be "Cannot"?

Reply:
Both are acceptable, but cannot is more acceptable, so yes...

Reply:
In addition to what's already been mentioned, I'll just make a couple of remarks here.

1. The `catch` clause for `std::runtime_error` should be listed before the `catch` clause for `std::exception`.  In the current approach, we'll never get `runtime_error`s because they'll all be swallowed by `exception`s.  On the other hand, since we actually perform the same actions in both cases, we might just remove the `runtime_error` handling altogether.
2. We don't actually know that we failed to "open" the file if we get here.  The only thing we do know is that we failed to read the header information. I think I'd prefer that the diagnostic message be a little more specific and **possibly** include `e.what()` as well.  For instance something along the lines of
    ```C++
    fmt::format("Unable to read array header from {}: {}", this->inputFilename, e.what())
    ```

Reply:
@bska: Thanks! I've included your changes.
@atgeirr: Should the part on MPI possibly be included in opm-models/opm/models/parallel/tasklets.hh?
I actually get this error when running flow sequentially (if there are already output files present where the header can't be read).